src/main/row.ts

```diff
@@ -76,58 +76,28 @@ export function generateIdForRow(row: Row): string {
  * @returns boolean representing whether a given Row has been assigned a unique id or not
  */
 export function hasId(row: Row): boolean {
-    return row.metadata.getValue() !== null;
+    const id: string | null = row.metadata.getValue();
+
+    return id !== null && id !== "";
```

Comment:
nasty, what case does it end up being ""?

Reply:
if you could add the reasoning in as a comment that may be helpful

Reply:
Same issue as describe here https://github.com/andrewc5716/onestop-basecamp-integration/blob/9d1ead0d3b674da2bb4cd600ec6ea23df6103ae1/src/main/row.ts#L51-L55
I'll add a comment to make this clear